public/app/features/dashboard-scene/scene/layout-responsive-grid/SceneCSSGridLayout.tsx

```diff
@@ -0,0 +1,343 @@
+
+  if (!currentLayoutIsActive && containerRef.current) {
+    containerRef.current.style.overflow = '';
+    containerRef.current.style.height = '';
+    containerRef.current.style.transition = '';
+  }
+
+  oldDropZone.current = dropZone;
+
+  useLayoutEffect(() => {
+    if (containerRef.current) {
+      model.setContainer(containerRef.current);
+    }
+  });
+
+  if (isHidden) {
+    return null;
+  }
+
+  return (
+    <div
+      className={styles.container}
+      ref={containerRef}
+      onPointerEnter={() => {
+        if (!containerRef.current || model === activeLayout || !model.dragManager) {
+          return;
+        }
+
+        model.dragManager.dropZones = [];
+        model.dragManager.setState({ activeLayout: model, dropZone: undefined });
+
+        // Probably a better way of doing this, but we have to wait for react to add the new placeholder
+        // before we calculate the drop zones
+        setTimeout(() => {
+          model.dragManager?.refreshDropZones();
+        }, 1);
+      }}
+    >
+      {children.map((item, i) => {
+        const Component = item.Component as ComponentType<ResponsiveGridItemRenderProps<SceneObject>>;
+        // const Wrapper = isLazy ? LazyLoader : 'div';
+        const Wrapper = 'div';
+        const isHidden = 'isHidden' in item.state && typeof item.state.isHidden === 'boolean' && item.state.isHidden;
+        const isDragging = item === activeItem;
+        if (isDragging) {
+          return null;
+        }
+
+        return (
+          <Wrapper key={item.state.key!} className={styles.itemWrapper} data-order={isHidden ? -1 : i}>
+            <Component model={item} />
+          </Wrapper>
+        );
+      })}
+      {currentLayoutIsActive && <div style={{ gridRow: dropZone?.rowIndex, gridColumn: dropZone?.columnIndex }}></div>}
+    </div>
+  );
+}
+
+const getStyles = (theme: GrafanaTheme2, state: SceneCSSGridLayoutState) => ({
+  container: css({
+    display: 'grid',
+    position: 'relative',
+    gridTemplateColumns: state.templateColumns,
+    gridTemplateRows: state.templateRows || 'unset',
+    gridAutoRows: state.autoRows || 'unset',
+    rowGap: theme.spacing(state.rowGap ?? 1),
+    columnGap: theme.spacing(state.columnGap ?? 1),
+    justifyItems: state.justifyItems || 'unset',
+    alignItems: state.alignItems || 'unset',
+    justifyContent: state.justifyContent || 'unset',
+    flexGrow: 1,
+
+    [theme.breakpoints.down('md')]: state.md
+      ? {
+          gridTemplateRows: state.md.templateRows,
+          gridTemplateColumns: state.md.templateColumns,
+          rowGap: state.md.rowGap ? theme.spacing(state.md.rowGap ?? 1) : undefined,
+          columnGap: state.md.columnGap ? theme.spacing(state.md.rowGap ?? 1) : undefined,
+          justifyItems: state.md.justifyItems,
+          alignItems: state.md.alignItems,
+          justifyContent: state.md.justifyContent,
+        }
+      : undefined,
```

Comment:
not sure we will use these in dashboards (think we will enforce good responsive templates instead)